code/modules/power/apc.dm

```diff
@@ -781,9 +781,72 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/power/apc/auto_name, APC_PIXEL_OFFSET
 		return
 	if(!user.canUseTopic(src, !issilicon(user)) || !isturf(loc))
 		return
-	togglelock(user)
+	if(!ishuman(user))
+		return
+	else
+		var/mob/living/carbon/human/H = user
+		var/obj/item/organ/stomach/maybe_stomach = H.getorganslot(ORGAN_SLOT_STOMACH)
+		if(!istype(maybe_stomach, /obj/item/organ/stomach/ethereal))
+			togglelock(user)
+		else
+			var/obj/item/organ/stomach/ethereal/stomach = maybe_stomach
+			if(stomach.crystal_charge >= ETHEREAL_CHARGE_NORMAL)
+				togglelock(user)
+			ethereal_interact(user,modifiers)
 	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
 
+/obj/machinery/power/apc/proc/ethereal_interact(mob/living/user,list/modifiers)
+	if(!ishuman(user))
+		return
+	else
```

Comment:
dude wtf- returning ends the proc, so why did you keep the else case. remove those and unindent every time you remove an else after the return

Reply:
its been a while okay, Its late, Im sorry